openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -0,0 +1,99 @@
+from rest_framework.response import Response
+from rest_framework import viewsets, status
+from rest_framework.decorators import permission_classes, authentication_classes
+from rest_framework.filters import OrderingFilter
+from rest_framework.authentication import SessionAuthentication
+
+from django.contrib.auth.models import User
+from student.models import UserAttribute
+from django.db.models.query_utils import Q
+
+from .serializers import UserSerializer
+
+from .permissions import CanAccessPakXAdminPanel
+
+from .constants import GROUP_TRAINING_MANAGERS, ADMIN, TRAINING_MANAGER
+
+
+@authentication_classes([SessionAuthentication])
+@permission_classes([CanAccessPakXAdminPanel])
+class UserProfileViewSet(viewsets.ModelViewSet):
+    queryset = None
+
+    serializer_class = UserSerializer
+
+    filter_backends = [OrderingFilter]
+
+    ordering = ['-id']
+
+    def create(self, request, *args, **kwargs):
+        if request.data['active_status'] is not None and request.data['active_status'] in [0, 1]:
+            return self.user_toggle_activation(bool(request.data["active_status"]), request.data["ids"])
+
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+
+    def destroy(self, request, *args, **kwargs):
+
+        if int(kwargs['pk']) in self.get_valid_ids():
+            User.objects.filter(id=kwargs['pk']).update(is_active=False)
```

Comment:
instead of getting valid IDs, make ORM query with org join to avoid the unnecessary queries and computations